src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -231,10 +231,18 @@ void MeshForwarder::UpdateIndirectMessages(void)
     }
 }
 
-otError MeshForwarder::EvictIndirectMessage(void)
+otError MeshForwarder::EvictMessage(uint8_t aPriority)
 {
     otError error = OT_ERROR_NOT_FOUND;
 
+#if OPENTHREAD_ENABLE_QOS
+    Message *message = mSendQueue.GetTail();
+    if (message->GetPriority() > aPriority)
+    {
+        RemoveMessage(*message);
```

Comment:
We also need to address the case where there are no more lower priority messages, but there are some messages with the same priority.  In that case, we need to pay attention to what indirect messages are queued and ensure that a given child or subset of children do not starve a given traffic class.  Even in the case where we have an incoming flood of high-priority traffic, we should handle that well.  The existing code below (when `OPENTHREAD_ENABLE_QOS` is `0` is not perfect, but it is simple and provides a good approximation.

For now, I think having the `else if (message->GetPriority() == aPriority)` perform the code below would be sufficient for this PR.

Reply:
Thanks for your explanation and suggestion. I will update the code.